### PR TITLE
Implement and Test Invitation Expiration Logic

### PR DESCRIPTION
**Pull request by nkzarrabi**
- Added the `is_expired` method to the `Invitation` model to check if an invitation has expired based on the `expires_at` field.
- Ensured personal invitations are deleted upon acceptance by integrating deletion logic within the `accept_invitation` method, wrapped in a transaction for atomicity.
- Created two new tests in `InvitationDeletionTests`:
  - `test_invitation_expiration` to verify that an invitation is correctly identified as expired when past its expiration date.
  - `test_invitation_deletion_after_accepting` to ensure personal invitations are deleted after being accepted, and open invitations remain unaffected.
- Adjusted the `User` model and related views/forms to handle the creation and acceptance of invitations, considering their expiration status.

### CODE REVIEW

File: core/forms.py [hunk 1]
```diff
@@ -56,10 +56,15 @@ class InvitationForm(forms.ModelForm):
         help_text='This message will be shown to the user you invite.',
         widget=forms.Textarea,
     )
+    is_open = forms.BooleanField(
```
Comment (nkzarrabi) [inline core/forms.py:59]: invitation form should feature open invite attribute. Not sure if this is the best way to do that

File: core/model_tests/test_invitations.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            )
+            # If we reach here, the invitation acceptance was successful
+            self.assertFalse(
+                invitation.is_expired(),
+                "The invitation should not be expired yet.")
+        except Exception as e:
+            # If an exception is raised, the test should fail
+            self.fail(f"Accepting the invitation raised an exception: {e}")
+
+    def test_invitation_deletion_after_accepting(self):
```
Comment (nkzarrabi) [inline core/model_tests/test_invitations.py:75]: The expiration check is currently done when the user either accepts within reasonable timeframe , which then causes the invite to be deleted 
Or 
the user tries to accept an invite after the invite period has passed, which prevents them from accepting a valid invite. 
So they would need to repeat the process if both parties were trying to connect.

File: core/models.py [hunk 3]
```diff
@@ -212,6 +217,9 @@ def accept_invitation(self, invitation, *, circles):
                 connection=connection.opposite,
             )
 
+        if not invitation.is_open:
```
Comment (nkzarrabi) [inline core/models.py:220]: deletion after success, assuming invitation is not expired and connection has been made.

File: core/models.py [hunk 4]
```diff
@@ -226,6 +234,23 @@ class Invitation(models.Model):
...
+
+    is_open = models.BooleanField(default=False)
+    expires_at = models.DateTimeField(null=True, blank=True)
+
+    def is_expired(self):
+        if not self.is_open and self.expires_at:
+            return timezone.now() > self.expires_at
+        return False
+
+    @property
```
Comment (nkzarrabi) [inline core/models.py:247]: I don't remember why I added the other stuff. 
Do you agree with the 
is_open attribute?
Correct expiration approach?

File: core/test_integration.py [hunk 5]
```diff
@@ -47,7 +47,6 @@ def tearDownClass(cls):
         super().tearDownClass()
 
         if getattr(cls, 'browser') is not None:
-            time.sleep(2)  # Keep the browser open for 2 seconds
             cls.browser.quit()
```
Comment (nkzarrabi) [inline core/test_integration.py:50]: Got rid of implicit wait. It was a debugging thing I used when running the browser.. non-headless? head? lol

File: core/views.py [hunk 6]
```diff
@@ -255,7 +256,16 @@ def get_form_kwargs(self):
         return result
 
     def form_valid(self, form):
```
Comment (nkzarrabi) [inline core/views.py:258]: handling expiration time in views when invite is created by the user. 
Perhaps I missed something with the timezone. Is there a different way to access this?
I've only seen the timezone/time stuff referenced in User.

**Review by nkzarrabi — COMMENTED** [review]
@kerkeslager any thoughts?

**Comment by kerkeslager** [comment]
@nkzarrabi This looks really good. The one thing I'd change is, let's not have an `expires_at` field. Instead, let's have a `created_utc` that tracks when the `Invitation` was created (take a look at `created_utc` on the `Connection` model for how to do that quite easily). Then we can create an `expires_at` property like:

```
@property
def expires_at(self):
    return self.created_utc + timedelta(days=7)
```

This allows us to change how long it takes for an invite to expire without having to migrate existing times saved in the database.

**Comment by kerkeslager** [comment]
Oh, I guess `DEFAULT_EXPIRATION` can be moved into `settings.py` too (probably with a name like `INVITE_LIFESPAN` to be more specific--`DEFAULT_EXPIRATION` is clear enough in the context of the class, but once moved to `settings.py`, that name would be unclear.

**Comment by nkzarrabi** [comment]
> @nkzarrabi This looks really good. The one thing I'd change is, let's not have an `expires_at` field. Instead, let's have a `created_utc` that tracks when the `Invitation` was created (take a look at `created_utc` on the `Connection` model for how to do that quite easily). Then we can create an `expires_at` property like:
> 
> ```
> @property
> def expires_at(self):
>     return self.created_utc + timedelta(days=7)
> ```
> 
> This allows us to change how long it takes for an invite to expire without having to migrate existing times saved in the database.

done!

**Pull request merged**